rust/emitter/src/ast_emitter.rs

```diff
@@ -119,6 +119,35 @@ impl AstEmitter {
         Err(EmitError::NotImplemented("TODO: this"))
     }
 
+    fn emit_if(&mut self, if_statement: &IfStatement) -> Result<(), EmitError> {
+        self.emit_expression(&if_statement.test)?;
+
+        let offset_alternate = self.emit.bytecode_offset();
```

Comment:
I prefer having emitter helper before writing multiple similar control structure (e.g. if vs conditional-expression),
but there's still a design decision needed for the emitter helper itself (how to utilize type system for enforcing correct usage),
so, for now, I think this is fine.